internal/cloud/realm/auth_test.go

```diff
@@ -78,16 +79,58 @@ func TestRealmAuthRefresh(t *testing.T) {
 		assert.Equal(t, nil, err)
 
 		// invalidate the session's tokens
-		session.AccessToken = ""
 		session.RefreshToken = session.AccessToken
+		session.AccessToken = ""
 
 		profile := mock.NewProfileWithSession(t, session)
 
 		client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
 		_, err = client.AuthProfile()
 		assert.Equal(t, realm.ErrInvalidSession{}, err)
 	})
-	// TODO: REALMC-7719 add test for expired credentials and test for ensuring profile cleared on invalid session
+
+	t.Run("with an expired access token", func(t *testing.T) {
+		u.SkipUnlessExpiredAccessTokenPresent(t)
+		t.Run("should use the refresh token to generate a new access token", func(t *testing.T) {
+			client := realm.NewClient(u.RealmServerURL())
+			u.SkipUnlessRealmServerRunning(t)
+			session, err := client.Authenticate(u.CloudUsername(), u.CloudAPIKey())
+			assert.Equal(t, nil, err)
+
+			session.AccessToken = u.ExpiredAccessToken() // Set the access token to an expired token
+
+			profile, teardown := mock.NewProfileFromTmpDir(t, "auth_refresh_test")
+			defer teardown()
+			profile.SetRealmBaseURL(u.RealmServerURL())
+			profile.SetSession(auth.Session{session.AccessToken, session.RefreshToken})
+
+			client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
+			_, err = client.AuthProfile()
+			assert.Nil(t, err)
+		})
+		t.Run("should error if both the access and refresh tokens are expired", func(t *testing.T) {
+			client := realm.NewClient(u.RealmServerURL())
+			u.SkipUnlessRealmServerRunning(t)
+			session, err := client.Authenticate(u.CloudUsername(), u.CloudAPIKey())
+			assert.Equal(t, nil, err)
+
+			session.AccessToken = u.ExpiredAccessToken()  // Set the access token to an expired token
+			session.RefreshToken = u.ExpiredAccessToken() // Set the refresh token to an random expired token
+
+			profile, teardown := mock.NewProfileFromTmpDir(t, "auth_refresh_test")
+			defer teardown()
+			profile.SetRealmBaseURL(u.RealmServerURL())
+			profile.SetSession(auth.Session{session.AccessToken, session.RefreshToken})
+
+			client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
+			_, err = client.AuthProfile()
+			assert.Equal(t, realm.ErrInvalidSession{}, err)
+
+			t.Log("and clear the session tokens")
+			assert.Equalf(t, "", profile.Session().AccessToken, "access token was not cleared")
```

Comment:
```suggestion
			assert.Equalf(t, "", profile.Session().AccessToken, "access token was not cleared: %q", profile.Session().AccessToken)
```

Reply:
I don't mind this change but I went with the above message just in case we don't want to print the auth tokens in the logs (not sure how big of a deal it is in this case). Thoughts?

Reply:
yeah thats fine then, lets leave it as is and out of the message here.  better safe than sorry